src/Equations/Fluid/HottailRateTermLowZ.cpp

```diff
@@ -0,0 +1,423 @@
+    
+    // Deriv. of Ec/Epar² 
+    real_t dEcOverEparSq_du = Box*Box* dInvOfEc_du / (j0*j0) + 2*Box* dBox_du / (j0*j0*Ec);
+    // Deriv. of Epar
+    real_t dEpar_du = j0*dEc_du / Box - Ec*j0*dBox_du / (Box*Box);
+    // Deriv. of dEpar/dt (since Epar_prev is constant)
+    real_t ddEpardt_du = dEpar_du / dt;
+    
+    // Deriv of lower limit in gamma integral sqrt(Ec/Epar)
+    real_t dlowlim_du = (dEc_du / sqrt(Ec*Epar[ir]) - sqrt(Ec)*dEpar_du / sqrt(Epar[ir]*Epar[ir]*Epar[ir]))/2;
+    derpar->lowlim = sqrt(Ec/Epar[ir]);
+    
+    // Evaluate f0 at lowlim
+    real_t f0_lowlim = derpar->dist->evaluateEnergyDistributionFromTau(ir,derpar->lowlim,derpar->tau);
+    // Integral of df0/du from lowlim to inf
+    real_t int_df0du = Intdf0du(derpar);
+    // Deriv. of integral in gamma with u dependance in both f0 and lowlim, acc. to Leibniz integral rule
+    real_t dGammaInt_du = -f0_lowlim * dlowlim_du + int_df0du;
+    
+    // dEpar/dt
+    real_t dotEpar = (Epar[ir] - Epar_prev[ir]) / dt;
+    if (dotEpar < 0){ // ensure non-negative runaway rate
+        dotEpar = 0;
+    }
+    
+    // Saved value of the integral in the expression for gamma
+    real_t gammaIntegral = integraloff0inGamma[ir];
+    // Deriv. of gamma wrt u, composed of sum from contributions by alpha, beta, delta
+    real_t dGamma_du = dEcOverEparSq_du * dotEpar * gammaIntegral + ddEpardt_du * gammaIntegral * Ec / (Epar[ir]*Epar[ir]) + dGammaInt_du * dotEpar * Ec / (Epar[ir]*Epar[ir]);
+    
+    return dGamma_du;
+    
+}
+
+
+/**
+ * Sets the Jacobian of this equation term
+ */
+bool HottailRateTermLowZ::SetJacobianBlock(const len_t /*uqtyId*/, const len_t derivId, FVM::Matrix *jac, const real_t*){
+    if(!HasJacobianContribution(derivId))
+        return false;
+
+    for(len_t ir=0; ir<nr; ir++){
+        real_t V = GetVolumeScaleFactor(ir);
+        if(derivId == id_ni){
+            len_t nZ  = ionHandler->GetNZ(); // Total number of ion types
+            const len_t *Zs = ionHandler->GetZs(); // Highest charge state for each ion type in order of ion type index
+            for(len_t iz = 0; iz<nZ; iz++){ // loop ion type index
+                for(len_t Z0=0; Z0<=Zs[iz]; Z0++){ // loop charge states
+                    len_t n = ionHandler->GetIndex(iz,Z0); // index in list [nD0 nD1 nNe0 ... nNe10] if D and Ne
+                    real_t dGamma = dGammadu(ir, derivId, n);
+                    jac->SetElement(ir, n*nr+ir, scaleFactor * dGamma * V);
+                }
+            }
+        } else {
+            real_t dGamma = dGammadu(ir, derivId, 0); // Independent of n
+            jac->SetElement(ir, ir, scaleFactor * dGamma * V);
+        }
+    }
+    return true;
+}
+
+
+/**
+ * Deallocator
+ */
+void HottailRateTermLowZ::Deallocate(){
+    if(Epar_prev != nullptr)
+        delete [] Epar_prev;
+        delete [] params;
+        delete [] integraloff0inGamma;
```

Comment:
The indentation here seems a bit misleading? Both ``params`` and ``integraloff0inGamma`` are outside of the ``if`` statement and will ALWAYS be deleted. Is this the intended behaviour?